### PR TITLE
Remove dual field constructors

### PR DESCRIPTION
**Pull request by nbenn**

### CODE REVIEW

File: tests/testthat/iris.csv
Comment (DivadNojnarg) [inline tests/testthat/iris.csv:1]: @nbenn Is this new file needed/expected?

**Review by DivadNojnarg — COMMENTED** [review]
TBD: update all the shinylive and Rmd code examples to comply with this new change.

Same as for #373, I can't test examples as the hidden field from data_block is causing a validation issue.

File: tests/testthat/iris.csv
Comment (nbenn) [reply tests/testthat/iris.csv:1]: Nope, my bad, sorry. I had experimented with [`withr::local_file()`](https://withr.r-lib.org/reference/with_file.html), but but somehow did not properly manage the mechanics of the API there. Replaced all with [`local_tempfile()`](https://withr.r-lib.org/reference/with_tempfile.html) and removed leftovers.

**Comment by DivadNojnarg** [comment]
I am currently checking, the plot layer vignette which needs to be updated since we pass data as parameter and we still have the 2 block constructors. We should have something like:

```r
new_ggplot_block <- function(...) {

  data_cols <- function(data) colnames(data)

  new_block(
    fields = list(
      x = new_select_field("wt", data_cols, type = "name"),
      y = new_select_field("qsec", data_cols, type = "name"),
      color = new_select_field("disp", data_cols, type = "name")
    ),
    expr = quote(
      ggplot(mapping = aes(x = .(x), y = .(y), color = .(color)))
    ),
    class = c("ggplot_block", "plot_block"),
    ...
  )
}
```

This applies to all vignettes still using our old approach.

**Pull request merged**